packages/x-data-grid-premium/src/components/promptControl/GridToolbarPromptControl.tsx

```diff
@@ -0,0 +1,272 @@
+import composeClasses from '@mui/utils/composeClasses';
+import useEventCallback from '@mui/utils/useEventCallback';
+import {
+  getDataGridUtilityClass,
+  GRID_CHECKBOX_SELECTION_FIELD,
+  gridColumnLookupSelector,
+  GridLogicOperator,
+  GridSingleSelectColDef,
+} from '@mui/x-data-grid';
+import { getValueOptions, getVisibleRows } from '@mui/x-data-grid/internals';
+import InputAdornment from '@mui/material/InputAdornment';
+import { DataGridPremiumProcessedProps } from '../../models/dataGridPremiumProps';
+import { GridApiPremium } from '../../models/gridApiPremium';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { PromptResponse } from '../../hooks/features/promptControl/types';
+import { RecordButton } from './RecordButton';
+
+type OwnerState = DataGridPremiumProcessedProps;
+
+const useUtilityClasses = (ownerState: OwnerState, recording: boolean) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['toolbarPromptControl', recording && 'toolbarPromptControl--recording'],
+    recordingIndicator: ['toolbarPromptControlRecordingIndicator'],
+    recordButton: ['toolbarPromptControlRecordButton'],
+    sendButton: ['toolbarPromptControlSendButton'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridToolbarPromptControlRoot = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'ToolbarPromptControl',
+  overridesResolver: (_props, styles) => styles.toolbarPromptControl,
+})<{ ownerState: OwnerState }>({
+  flex: 1,
+  display: 'flex',
+  flexDirection: 'row',
+});
+
+function sampleData(
+  apiRef: React.MutableRefObject<GridApiPremium>,
+  rootProps: DataGridPremiumProcessedProps,
+) {
+  const columns = Object.values(gridColumnLookupSelector(apiRef));
+  const rows = rootProps.rows;
+  const columnExamples: Record<string, any[]> = {};
+
+  columns.forEach((column) => {
+    columnExamples[column.field] = rows.slice(0, 5).map(() => {
+      const row = rows[Math.floor(Math.random() * rows.length)];
+      if (column.valueGetter) {
+        return column.valueGetter(row[column.field] as never, row, column, apiRef);
+      }
+      return row[column.field];
+    });
+  });
+
+  return columnExamples;
+}
+
+function generateContext(
+  apiRef: React.MutableRefObject<GridApiPremium>,
+  examples?: Record<string, any[]>,
+) {
+  const columns = Object.values(gridColumnLookupSelector(apiRef));
+  const columnsContext = columns.map((column) => ({
+    field: column.field,
+    description: column.description ?? null,
+    examples: examples?.[column.field] ?? column.unstable_examples ?? [],
+    type: column.type ?? 'string',
+    allowedOperators: column.filterOperators?.map((operator) => operator.value) ?? [],
+  }));
+
+  return `${apiRef.current.getLocaleText('toolbarPromptControlColumnsContextIntro')}\n${JSON.stringify(columnsContext)}`;
```

Comment:
Why use a locale text here?

Reply:
This is a static text providing some context for the columns. I was thinking that, in case someone uses a model in another language, they will need a different string here.
This is the english string used here
`The columns are described by the following JSON:`

Reply:
Right, I wonder if the language matters in this context 🤔
@romgrk What do you think?

Reply:
After checking this, I see that this part ends up as a system input in our backend which is in english.
I will do some testing to see if the languages can be mixed. For now, this string will not be translatable